src/widergb.rs

```diff
@@ -0,0 +1,273 @@
+//! # WideRgb Module
+//!
+//! This module provides the `WideRgb` struct, a representation of a color stimulus using unconstrained
+//! Red, Green, and Blue (RGB) floating-point values within a specified RGB color space. Unlike typical
+//! RGB values that are limited to the range `[0.0, 1.0]`, `WideRgb` allows values to extend beyond this
+//! range, accommodating out-of-gamut colors that cannot be accurately rendered by most devices.
+//!
+//! ## Key Features
+//! - **Unconstrained RGB Values:** The `WideRgb` struct stores RGB values that can extend beyond the
+//!   standard range, allowing for high-dynamic-range (HDR) and out-of-gamut colors.
+//! - **Color Space Support:** Supports various RGB color spaces, including sRGB, Adobe RGB, and custom
+//!   spaces, specified through the `RgbSpace` type.
+//! - **Observer Customization:** Allows the use of different colorimetric observers (e.g., CIE 1931,
+//!   CIE 2015), enhancing accuracy in color conversions and comparisons.
+//! - **Conversion Methods:** Provides methods to clamp, compress, and convert `WideRgb` values to
+//!   standard `Rgb` values within the `[0.0, 1.0]` range.
+//! - **Compatibility:** Includes conversions to `XYZ` tristimulus values, standard `Rgb`, and
+//!   `u8` arrays for 8-bit encoded RGB values.
+//!
+//! ## Example Usage
+//!
+//! ```rust
+//! use colorimetry::widergb::WideRgb;
+//!
+//! let wide_rgb = WideRgb::new(1.5, -0.2, 0.8, None, None);
+//! let compressed_rgb = wide_rgb.compress();
+//! let clamped_rgb = wide_rgb.clamp();
+//!
+//! println!("Compressed: {:?}", compressed_rgb.values());
+//! println!("Clamped: {:?}", clamped_rgb.values());
+//! ```
+//!
+//! ## Notes
+//! - The `compress` method applies chroma reduction and luminance scaling to ensure values fit within
+//!   the `[0.0, 1.0]` range, resulting in a lossy conversion.
+//! - The `clamp` method simply restricts values to the valid range without chroma adjustment, potentially
+//!   altering perceived color balance.
+//!
+//! ## Testing
+//! Unit tests are provided to verify the accuracy of conversions, clamping, and compression operations.
+
+use crate::{
+    colorant::Colorant,
+    data::observers::CIE1931,
+    illuminant::Illuminant,
+    observer::Observer,
+    rgb::{self, Rgb},
+    rgbspace::RgbSpace,
+    spectrum::Spectrum,
+    stimulus::Stimulus,
+    traits::{Filter, Light},
+    xyz::XYZ,
+};
+use approx::AbsDiffEq;
+use nalgebra::{Matrix3, Vector3};
+use std::{borrow::Cow, sync::OnceLock};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[wasm_bindgen]
+#[derive(Debug, Clone, Copy, PartialEq)]
+/// Represents a color stimulus using unconstrained Red, Green, and Blue (RGB) floating-point values
+/// within a device's RGB color space. The values can extend beyond the typical 0.0 to 1.0 range,
+/// allowing for out-of-gamut colors that cannot be accurately represented by the device.
+///
+/// # Fields
```

Comment:
A bit nitpicky. But since the fields are not public, documenting them in the public API docs can be confusing, and does not really help. My recommendation would be to document each field in doc comments attach to each field (which you already  have also, so that's fine!). The documentation showing up in the built docs should IMO only concern the public parts of the API.